src/components/ReservedUserData/ReservedUserData.tsx

```diff
@@ -25,6 +26,10 @@ interface Props {
 type StudentResults = { allRecords: number; data: ReservedStudent[] };
 
 export const ReservedUserData = () => {
+
+    const navigate = useNavigate();
+    const hrId = localStorage.getItem('userid');
```

Comment:
Łatwiej byłoby zrobić context,  które mają dostęp wszystkie komponenty i tam pobierać dane z localStorage będzie to mniej kodu i wygodniejsze w wprowadzaniu zmian